wasm/javascript-v8/module.go

```diff
@@ -0,0 +1,68 @@
+package v8
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+
+	"github.com/streamingfast/substreams/wasm"
+	"rogchap.com/v8go"
+)
+
+type V8Module struct {
+	iso      *v8go.Isolate
+	code     []byte
+	registry *wasm.Registry
+}
+
+//go:embed runtime/polyfill.bundle.js
+var polyfillCode string
+
+//go:embed runtime/prelude.bundle.js
+var preludeCode string
+
+func (mod *V8Module) NewInstance(ctx context.Context) (wasm.Instance, error) {
+	return NewV8Instance(mod.iso)
+}
+
+func (mod *V8Module) ExecuteNewCall(
+	ctx context.Context,
+	call *wasm.Call,
+	cachedInstance wasm.Instance,
+	arguments []wasm.Argument,
+	argValues map[string][]byte,
+) (wasm.Instance, error) {
+	var inst *V8Instance
+	var err error
+
+	if cachedInstance != nil {
+		inst = cachedInstance.(*V8Instance)
+	} else {
+		inst, err = NewV8Instance(mod.iso)
+		if err != nil {
+			return nil, fmt.Errorf("creating new V8 instance: %w", err)
+		}
+	}
+
+	if _, err = inst.ctx.RunScript(polyfillCode, "polyfill.js"); err != nil {
+		inst.Close(ctx)
+		return nil, fmt.Errorf("executing polyfill: %w", err)
+	}
+
+	if _, err = inst.ctx.RunScript(preludeCode, "prelude.js"); err != nil {
+		inst.Close(ctx)
+		return nil, fmt.Errorf("executing prelude: %w", err)
+	}
+
+	if _, err = inst.ctx.RunScript(string(mod.code), "bundle.js"); err != nil {
+		inst.Close(ctx)
+		return nil, fmt.Errorf("executing JS bundle: %w", err)
+	}
+
+	return inst, nil
```

Comment:
Who is closing the `inst` when there was no error within the script?

Golang as a `defer` functionality to perform an operation after the function finish but before returning to the caller. You duplicated the `inst.Close(...)` call which is error prone. 

Instead, remove all the manual call, and at line 44, do:

```go
		inst, err = NewV8Instance(mod.iso)
		if err != nil {
			return nil, fmt.Errorf("creating new V8 instance: %w", err)
		}
		defer func () { if err := inst.Close(); err != nil { /* log error */} }()
```

Reply:
After review, the instance is used and lifecycle is handled later, this comment does not appy.